client/composables/useSaveChangesSnackbar.ts

```diff
@@ -0,0 +1,65 @@
+import type { EventBusKey } from '@vueuse/core'
+
+const SaveChangesOpenEventKey: EventBusKey<undefined> = Symbol('save-changes-service-open')
+const SaveChangesCloseEventKey: EventBusKey<undefined> = Symbol('save-changes-service-close')
+const SaveChangesResetEventKey: EventBusKey<undefined> = Symbol('save-changes-service-reset')
+const SaveChangesSubmitEventKey: EventBusKey<undefined> = Symbol('save-changes-service-submit')
+const SaveChangesSubmitCompleteEventKey: EventBusKey<undefined> = Symbol('save-changes-service-submit-complete')
+
+export function useSubscribeSaveChangesEvent(
+  openListener: () => void,
+  closeListener: () => void,
+  completeListener: () => void,
+) {
+  const saveChangesOpenEvent = useEventBus(SaveChangesOpenEventKey)
+  const saveChangesCloseEvent = useEventBus(SaveChangesCloseEventKey)
+  const saveChangesResetEvent = useEventBus(SaveChangesResetEventKey)
+  const saveChangesSubmitEvent = useEventBus(SaveChangesSubmitEventKey)
+  const saveChangesSubmitCompleteEvent = useEventBus(SaveChangesSubmitCompleteEventKey)
+
+  onMounted(() => {
+    saveChangesOpenEvent.on(openListener)
+    saveChangesCloseEvent.on(closeListener)
+    saveChangesSubmitCompleteEvent.on(completeListener)
+  })
+
+  onBeforeUnmount(() => {
+    saveChangesOpenEvent.off(openListener)
+    saveChangesCloseEvent.off(closeListener)
+    saveChangesSubmitCompleteEvent.off(completeListener)
+  })
+
+  return { saveChangesSubmitEvent, saveChangesResetEvent } as const
```

Comment:
consistency: the useSaveChangesSnackbar returns functions which emits the events, while this one returns the event handlers themselves; chose one or the other.